make/Images.gmk

```diff
@@ -250,52 +250,80 @@ endif
-    SRC := $(ASYNC_PROFILER_IMPORT_PATH)/bin, \
-    DEST := $(JRE_IMAGE_DIR)/bin, \
-    FILES := asprof, \
-))
+ifeq ($(call isTargetOs, linux macosx)+$(ASYNC_PROFILER_IMPORT_ENABLED), true+true)
 
-$(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LIB_TO_JDK, \
-    SRC := $(ASYNC_PROFILER_IMPORT_PATH)/lib, \
-    DEST := $(JDK_IMAGE_DIR)/lib, \
-    FILES := async-profiler.jar converter.jar libasyncProfiler$(SHARED_LIBRARY_SUFFIX), \
-))
+  $(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_BIN_TO_JDK, \
+      SRC := $(ASYNC_PROFILER_IMPORT_PATH)/bin, \
+      DEST := $(JDK_IMAGE_DIR)/bin, \
+      FILES := asprof, \
+  ))
 
-$(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LIB_TO_JRE, \
-    SRC := $(ASYNC_PROFILER_IMPORT_PATH)/lib, \
-    DEST := $(JRE_IMAGE_DIR)/lib, \
-    FILES := async-profiler.jar converter.jar libasyncProfiler$(SHARED_LIBRARY_SUFFIX), \
-))
+  $(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_BIN_TO_JRE, \
+      SRC := $(ASYNC_PROFILER_IMPORT_PATH)/bin, \
+      DEST := $(JRE_IMAGE_DIR)/bin, \
+      FILES := asprof, \
+  ))
 
-$(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LEGAL_TO_JDK, \
-    SRC := $(ASYNC_PROFILER_IMPORT_PATH), \
-    DEST := $(JDK_IMAGE_DIR)/legal/async, \
-    FILES := CHANGELOG.md LICENSE README.md, \
-))
+  $(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LIB_TO_JDK, \
+      SRC := $(ASYNC_PROFILER_IMPORT_PATH)/lib, \
+      DEST := $(JDK_IMAGE_DIR)/lib, \
+      FILES := async-profiler.jar converter.jar libasyncProfiler$(SHARED_LIBRARY_SUFFIX), \
+  ))
 
-$(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LEGAL_TO_JRE, \
-    SRC := $(ASYNC_PROFILER_IMPORT_PATH), \
-    DEST := $(JRE_IMAGE_DIR)/legal/async, \
-    FILES := CHANGELOG.md LICENSE README.md, \
-))
+  $(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LIB_TO_JRE, \
+      SRC := $(ASYNC_PROFILER_IMPORT_PATH)/lib, \
+      DEST := $(JRE_IMAGE_DIR)/lib, \
+      FILES := async-profiler.jar converter.jar libasyncProfiler$(SHARED_LIBRARY_SUFFIX), \
+  ))
+
+  $(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LEGAL_TO_JDK, \
+      SRC := $(ASYNC_PROFILER_IMPORT_PATH), \
+      DEST := $(JDK_IMAGE_DIR)/legal/async, \
+      FILES := CHANGELOG.md LICENSE README.md, \
+  ))
 
+  $(eval $(call SetupCopyFiles, COPY_ASYNC_PROFILER_LEGAL_TO_JRE, \
+      SRC := $(ASYNC_PROFILER_IMPORT_PATH), \
+      DEST := $(JRE_IMAGE_DIR)/legal/async, \
+      FILES := CHANGELOG.md LICENSE README.md, \
+  ))
 
-JDK_TARGETS += $(COPY_ASYNC_PROFILER_BIN_TO_JDK) $(COPY_ASYNC_PROFILER_LIB_TO_JDK) $(COPY_ASYNC_PROFILER_LEGAL_TO_JDK)
-JRE_TARGETS += $(COPY_ASYNC_PROFILER_BIN_TO_JRE) $(COPY_ASYNC_PROFILER_LIB_TO_JRE) $(COPY_ASYNC_PROFILER_LEGAL_TO_JRE)
+  JDK_TARGETS += $(COPY_ASYNC_PROFILER_BIN_TO_JDK) $(COPY_ASYNC_PROFILER_LIB_TO_JDK) $(COPY_ASYNC_PROFILER_LEGAL_TO_JDK)
+  JRE_TARGETS += $(COPY_ASYNC_PROFILER_BIN_TO_JRE) $(COPY_ASYNC_PROFILER_LIB_TO_JRE) $(COPY_ASYNC_PROFILER_LEGAL_TO_JRE)
 
-endif
+  ifeq ($(call isTargetOs, macosx), true)
```

Comment:
I'm wondering if we could do the signing during file copy in a special install-file macro?

See: https://github.com/SAP/SapMachine/blob/06f965bc6f0407f9a2094e507b9ffe9405e5992d/make/common/CopyFiles.gmk#L69
and:
https://github.com/SAP/SapMachine/blob/06f965bc6f0407f9a2094e507b9ffe9405e5992d/make/common/FileUtils.gmk#L126

So maybe we can define an install-file-and-sign that we would use for the binaries?

Reply:
For just one exe and one shared lib it is not really worth the effort ; if we would have 5 or 10 why not